### PR TITLE
DOCSUP-1675 Docs for the cast_keep_nullable setting

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #11733

### CODE REVIEW

**Comment by blinkov** [comment]
converstion => conversion

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -1899,4 +1899,53 @@ Possible values:
 
 Default value: `120`.
 
+## cast_keep_nullable {#cast_keep_nullable}
+
+Enables or disables converstion to nullable types for [CAST](../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) operator.
```
Comment (BayoNet) [inline docs/en/operations/settings/settings.md:1904]:
```suggestion
Enables or disables keeping of the `Nullable` data type in [CAST](../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) operations.
```

Comment (BayoNet) [reply docs/en/operations/settings/settings.md:1904]: Also we need a link to Nullable.

File: docs/en/operations/settings/settings.md [hunk 2]
```diff
@@ -1899,4 +1899,53 @@ Possible values:
...
+## cast_keep_nullable {#cast_keep_nullable}
+
+Enables or disables converstion to nullable types for [CAST](../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) operator. 
+
+If set, `CAST(something_nullable AS Type)` returns `Nullable(Type)`.
+
+Possible values:
+
+-   0 — Nullable type converstion disabled.
+-   1 — Nullable type converstion enabled.
```
Comment (BayoNet) [inline docs/en/operations/settings/settings.md:1911]:
```suggestion
-  0 — The final type of `CAST` exactly the destination data type specified.
-  1 — The final type of `CAST` becomes `Nullable(DestinationDataType).
```

File: docs/en/operations/settings/settings.md [hunk 3]
```diff
@@ -1899,4 +1899,53 @@ Possible values:
...
+
+Default value: `0`.
+
+**Examples** 
+
+Query with nullable type converstion disabled:
```
Comment (BayoNet) [inline docs/en/operations/settings/settings.md:1917]:
```suggestion
The following query exactly results in the destination data type:
```

File: docs/en/operations/settings/settings.md [hunk 4]
```diff
@@ -1899,4 +1899,53 @@ Possible values:
...
+
+Result:
+
+```text
+┌─x─┬─toTypeName(CAST(toNullable(toInt32(0)), 'Int32'))─┐
+│ 0 │ Int32                                             │
+└───┴───────────────────────────────────────────────────┘
+```
+
+Query with nullable type converstion enabled:
```
Comment (BayoNet) [inline docs/en/operations/settings/settings.md:1932]:
```suggestion
The following query results in the `Nullable` modification on the destination data type:
```

File: docs/ru/operations/settings/settings.md [hunk 5]
```diff
@@ -1769,4 +1769,53 @@ SELECT idx, i FROM null_in WHERE i IN (1, NULL) SETTINGS transform_null_in = 1;
 
 Значение по умолчанию: `120` секунд.
 
+## cast_keep_nullable {#cast_keep_nullable}
+
+Включает или отключает финальное преобразование аргумента функции [CAST](../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) к типу `Nullable`.
```
Comment (alexey-milovidov) [inline docs/ru/operations/settings/settings.md:1774]: Not clear enough.

File: docs/ru/operations/settings/settings.md [hunk 6]
```diff
@@ -1769,4 +1769,53 @@ SELECT idx, i FROM null_in WHERE i IN (1, NULL) SETTINGS transform_null_in = 1;
...
+
+Если настройка включена, то функция `CAST(something_nullable AS Type)` возвращает `Nullable(Type)`.
```
Comment (alexey-milovidov) [inline docs/ru/operations/settings/settings.md:1776]: Plain wrong.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

**Comment by alexey-milovidov** [comment]
The setting allow to *keep nullable* if the data type is already nullable.

**Comment by alexey-milovidov** [comment]
Please make some experiments and edit the text...

File: docs/ru/operations/settings/settings.md (hunk 6 above)
Comment (olgarev) [reply docs/ru/operations/settings/settings.md:1887]: I updated the text.

Comment (BayoNet) [reply docs/ru/operations/settings/settings.md:1887]: You have updated the text from the previous commentary :-). 

The latter commentary from Alexey clarifies the situation: "The setting allows to keep nullable if the data type is already nullable." This means that the sentence "Если настройка включена, то функция `CAST(something_nullable AS Type)` возвращает `Nullable(Type)`." is really wrong.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**